documentation/src/main/asciidoc/mapping.asciidoc

```diff
@@ -309,6 +309,67 @@ Should you want to make a property sortable but not searchable, still an `@Field
 bridge configuration can be inherited). It can be marked with `store = Store.NO` and `index = Index.NO`, causing
 only the doc value field required for sorting to be added, but not a regular index field.
 
+Fields added through class-level bridges or custom field-level bridges (when not using the default field name) cannot
+be marked as sortable by means of the `@SortableField` annotation. Instead the field bridge itself has to add the
+required doc value fields, in addition to the document fields it adds. Furthermore such bridge needs to implement the
+`MetadataProvidingFieldBridge` interface which defines a method `getSortableFieldNames()` for returning the names of
+the sortable fields created by this bridge:
+
+[[example-adding-docvaluefield]]
+.Adding a sortable field via a custom field bridge
+====
+[source, JAVA]
+----
+public class MyClassBridge implements MetadataProvidingFieldBridge {
+
+    @Override
+    public void set(
+            String name, Object value, Document document, LuceneOptions luceneOptions) {
+
+        // add other document field(s)
+        // ...
+
+        // add doc value field
+        document.add( new NumericDocValuesField( "myField", 311 );
+    }
+
+    Set<String> getSortableFieldNames(String name) {
+        return Collections.singleton( "myField" );
+    }
+}
+----
+====
+
+The names returned from `getSortableFieldNames()` will be used for sort validation upon query execution. The name
+passed to the method is the default field name also passed to `set()`. It needs to be used consistently with `set()`,
+e.g. as a prefix for all custom fields added.
+
+[NOTE]
+====
+The `MetadataProvidingFieldBridge` contract is under active development and considered experimental at this time. It
+may be altered in future revisions, e.g. by adding further methods, thus breaking existing implementations.
+====
+
+[[flagging-uncovered-sorts]]
+====== Flagging uncovered sorts
+By default Hibernate Search will transparently create an uninverting index reader when running a query with sorts not
+covered by the sortable fields configured as described above. While this allows to execute the query, relying on index
+uninverting negatively impacts performance.
+
+You thus can optionally advice Hibernate Search to raise an exception when detecting uncovered sorts. To do so, specify
+the following option:
+
+.Disabling automatic index uninverting for uncovered sorts
+====
+[source]
+----
+hibernate.search.index_uninverting_allowed = false
+----
+====
+
+You e.g. may set this to "false" during testing to identify the sortable fields required for your queries and set it to
+"true" in production environments to fall back to index uninverting for uncovered sorts accidentally left over.
```

Comment:
> By the way, is a warning logged somewhere instead of an exception in case the property is not set?

RIght, the javadoc tells that there is warning. Maybe we write it in the documentation as well